File: leetcode/solutions/0678.ValidParenthesisString/checkValidString_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import checkValidString from "./checkValidString.ts";

Deno.test("0678. Valid Parenthesis String", () => {
  type test = [string, boolean];
  const tests: test[] = [
    ["", true],
    ["(", false],
    [")", false],
    ["()", true],
    ["()()", true],
    [")*(", false],
    ["*(()", false],
    ["(*)", true],
    ["(*))", true],
    ["((*)", true],
    ["()*)*)", true],
    ["())", false],
    ["*)**", true],
    ["(*()", true],
  ];

  for (const [arg, want] of tests) {
    assertEquals(checkValidString(arg), want);
  }
});

File: leetcode/solutions/0678.ValidParenthesisString/checkValidString_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_0678_ValidParenthesisString(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  string
		out bool
	}{
		{"", true},
		{"(", false},
		{")", false},
		{"()", true},
		{"()()", true},
		{")*(", false},
		{"*(()", false},
		{"(*)", true},
		{"(*))", true},
		{"((*)", true},
		{"()*)*)", true},
		{"())", false},
		{"*)**", true},
		{"(*()", true},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := checkValidString(tt.in)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}
